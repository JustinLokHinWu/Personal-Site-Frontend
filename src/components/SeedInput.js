import { Input } from 'antd'
import "antd/dist/antd.css";

const SeedInput = ({seed, setSeed}) => {
    const onChange = (e) => {
        const value = e.target.value
        const reg = /^[0-9]*$/
        if (value === null || value === '') {
            setSeed(null)
        }
        else if (reg.test(value) && +value < Number.MAX_SAFE_INTEGER){
            setSeed(value)
        }
    }

    return (
        <Input
            placeholder='Random'
            value={seed}
            style={{ width: 150 }}
            onChange={onChange}
            />
    )
}

export default SeedInput
