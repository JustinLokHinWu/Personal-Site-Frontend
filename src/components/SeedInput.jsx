import PropTypes from 'prop-types';
import React from 'react';
import { Input } from 'antd';

function SeedInput({ seed, setSeed }) {
  const onChange = (e) => {
    const { value } = e.target;
    const reg = /^[0-9]*$/;
    if (value === null || value === '') {
      setSeed(null);
    } else if (reg.test(value) && +value < Number.MAX_SAFE_INTEGER) {
      setSeed(value);
    }
  };

  return (
    <Input
      name="SeedInputForm"
      placeholder="Random"
      value={seed}
      onChange={onChange}
    />
  );
}

SeedInput.defaultProps = {
  seed: null,
};

SeedInput.propTypes = {
  seed: PropTypes.number,
  setSeed: PropTypes.func.isRequired,
};

export default SeedInput;
