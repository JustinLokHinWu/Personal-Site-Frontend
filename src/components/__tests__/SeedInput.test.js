// import { useState } from 'react'
import React from 'react';
import renderer from 'react-test-renderer';
import {render, screen, getByText, fireEvent} from '@testing-library/react'
import SeedInput from '../SeedInput';


test('SeedInput snapshot should match', () => {
    expect(1+2).toBe(3);
    
    const seed = null
    const setSeed = jest.fn()
    const component = renderer.create(
        <SeedInput seed={seed} setSeed={setSeed} /> 
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot()
})

// test('Seed should be null when input is null', () => {
//     const seed = null
//     const setSeed = jest.fn()
//     render(
//         <SeedInput seed={seed} setSeed={setSeed} />
//     )


//     // const component = renderer.create(
//     //     <SeedInput seed={seed} setSeed={setSeed} /> 
//     // )
//     // screen.render(
//     //     <SeedInput seed={seed} setSeed={setSeed} /> 
//     // )
//     const a = screen.getBy(component, 'SeedInputForm')
//     const b = null
// })
