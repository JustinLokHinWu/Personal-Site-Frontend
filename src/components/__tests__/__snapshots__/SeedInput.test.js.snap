// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeedInput snapshot should match 1`] = `
<input
  className="ant-input"
  name="SeedInputForm"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  placeholder="Random"
  type="text"
  value=""
/>
`;
